nca/FWRules/MinimizeFWRules.py

```diff
@@ -668,7 +668,7 @@ def get_fw_rules_in_required_format(self, add_txt_header=True, add_csv_header=Tr
             query_name += ', config: ' + self.output_config.configName
         output_format = self.output_config.outputFormat
         if output_format not in FWRule.supported_formats:
-            print(f'error: unexpected outputFormat in output configuration value [should be txt/yaml/csv],  '
+            print(f'error: unexpected outputFormat in output configuration value [should be txt/yaml/csv/md/json],  '
```

Comment:
missing the new `txt_no_fw_rules` format here.
is it possible to get the supported formats from where they are defined instead of duplicating them here?
This duplication is error-prone.

Reply:
I didn't add the `txt_no_fw_rules` intentionally, this `FWRules def` is not reachable for` txt_no_fw_rules, dot, jpg `formats (these formats don't use the FWrules/ minimize them)

Reply:
But if I have a typo in the specified output format , nca will reach this error, even though it should not necessarily be a fw-rules format. can it be checked/caught earlier, so the error message displays all possible output formats?

Reply:
I see, you are right 
only when returning from the "query.execute" will get the message of `<format with typo> output format is not supported for this query`
I may list all formats again here (even if the format does not use the fw_rules)
or might call this def only if the format supports FW-rule (add if statements in the relevant queries calling this def)

Reply:
I think it may be preferred to check even before the query is executed

Reply:
previously, we executed the query regardless of the `outputFormat `(an error was raised only if it gets to `MinimizeFWRules`)
In last commit if the format is not supported , the query will not be executed, and return with the not_executed flag so the relevant message of `outputConfiguration`:  `<format with typo> output format is not supported for this query` will be printed